app/services/supplier_chooser.rb

```diff
@@ -1,14 +1,15 @@
 class SupplierChooser
-  def initialize(doorkeeper_application_owner, from_location)
-    @doorkeeper_application_owner = doorkeeper_application_owner
-    @from_location = from_location
+  attr_accessor :effective_date, :location
+
+  def initialize(effective_date, location)
+    self.effective_date = effective_date
+    self.location = location
   end
 
   def call
-    return @doorkeeper_application_owner if @doorkeeper_application_owner
+    return unless effective_date.present? && location.present?
 
-    # TODO: Replace with the frontend supplied supplier
-    #       There can only ever be one supplier per location, currently
-    @from_location.suppliers.first
+    supplier_location = SupplierLocation.location(location.id).effective_on(effective_date).first
```

Comment:
is there a chance we might have multiple results returned?

Reply:
Potentially, although not intentionally. So this picks one arbitrarily if there are multiple results - which would indeed indicate overlapping effective dates and possibly multiple suppliers that can service the location. The business rule however is that only one supplier serves a location during the contract period.

Reply:
doesn't have to be this pr, but something to think about regarding enforcing this rule to have one either in validations or the rake task

Reply:
yep, good point: worth adding a comment about this line, e.g. `# NB: business rules _should_ preclude multiple suppliers` for visibility